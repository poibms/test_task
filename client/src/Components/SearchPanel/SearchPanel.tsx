import React, { useState, FC } from 'react';
import './SearchPanel.css';

interface ISearchProps {
	onSubmitByName: (value: string) => void;
	error: boolean;
}

const SearchPanel: FC<ISearchProps> = ({
	onSubmitByName,
	error,
}: ISearchProps) => {
	const [value, setValue] = useState('');

	const submitForm = (event: React.FormEvent<HTMLFormElement>) => {
		event.preventDefault();
		onSubmitByName(value);
		setValue('');
	};
	let inputStyle = { border: '3px solid #00B4CC' };
	if (error) {
		inputStyle = { border: '3px solid #c10020' };
	}

	return (
		<form className="search-form" onSubmit={submitForm}>
			<input
				className="onChange"
				type="text"
				placeholder="Enter city"
				value={value}
				style={inputStyle}
				onChange={(e) => setValue(e.currentTarget.value)}
			/>
			<input type="submit" className="onSearch" value="Search" />
		</form>
	);
};

export default SearchPanel;
