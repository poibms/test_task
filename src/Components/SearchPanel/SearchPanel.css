.search-form {
    display: flex;
    width: 400px;
    box-shadow: 0 5px 10px 0 rgb(0 0 0 / 10%);
}

.search-form input[type="text"]  {
    width: 100%;
    padding: 5px;
    height: 20px;
    border-radius: 5px 0 0 5px;
    color: #00B4CC;
    outline: none;
}

.search-form input[type="submit"] {
    min-width: 60px;
    border: 1px solid #00B4CC;
    background: #00B4CC;
    text-align: center;
    color: #fff;
    border-radius: 0 5px 5px 0;
    cursor: pointer;
    font-size: 14px;
}