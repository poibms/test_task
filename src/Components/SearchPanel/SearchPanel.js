import React, { Component } from 'react';
import './SearchPanel.css';

export default class SearchPanel extends Component {
	constructor(props) {
		super(props);
		this.state = {
			value: '',
		};
		this.onChangeSearchInput = this.onChangeSearchInput.bind(this);
		this.onSubmitSearchForm = this.onSubmitSearchForm.bind(this);
	}

	onSubmitSearchForm(e) {
		e.preventDefault();
		const { value } = this.state;
		this.props.onSubmitByName(value);
		this.setState({
			value: '',
		});
	}

	onChangeSearchInput(event) {
		this.setState({
			value: event.target.value,
			// value: event,
		});
	}

	render() {
		const { value } = this.state;

		return (
			<form className="search-form" onSubmit={this.onSubmitSearchForm}>
				<input
					className="onChange"
					type="text"
					placeholder="Enter city"
					value={value}
					onChange={this.onChangeSearchInput}
				/>
				<input type="submit" className="onSearch" />
				{/* <button type="button" className="click" /> */}
			</form>
		);
	}
}
