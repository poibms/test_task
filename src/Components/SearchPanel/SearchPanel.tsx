import React, { useState, FC } from 'react';
import './SearchPanel.css';

interface ISearchProps {
	onSubmitByName: (value: string) => void;
}

const SearchPanel: FC<ISearchProps> = ({ onSubmitByName }: ISearchProps) => {
	const [value, setValue] = useState('');

	const submitForm = (event: React.FormEvent<HTMLFormElement>) => {
		event.preventDefault();
		onSubmitByName(value);
		// setValue('');
	};

	return (
		<form className="search-form" onSubmit={submitForm}>
			<input
				className="onChange"
				type="text"
				placeholder="Enter city"
				value={value}
				onChange={(e) => setValue(e.currentTarget.value)}
			/>
			<input type="submit" className="onSearch" value="Search" />
		</form>
	);
};

export default SearchPanel;
