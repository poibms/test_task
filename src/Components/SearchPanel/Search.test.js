import { configure, shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import SearchPanel from './SearchPanel';

configure({ adapter: new Adapter() });

const setUp = shallow(<SearchPanel />);
describe('Search component', () => {
	const mockCallBack = jest.fn();
	let component;
	let instance;

	beforeEach(() => {
		component = setUp();
		instance = component.instance();
	});
	it('snapShot component', () => {
		expect(component).toMatchSnapshot();
	});
	it.skip('testing click', () => {
		expect(mockCallBack.mock.calls.length).toBe(0);
		component.find('.onSearch').simulate('click');
		expect(mockCallBack.mock.calls.length).toBe(1);
	});

});
