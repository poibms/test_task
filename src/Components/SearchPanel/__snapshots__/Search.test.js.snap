// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component snapShot component 1`] = `ShallowWrapper {}`;
